assembl/graphql/discussion.py

```diff
@@ -647,6 +644,15 @@ def mutate(root, args, context, info):
             if with_moderation is not None:
                 discussion.preferences['with_moderation'] = with_moderation
 
+            if slug != discussion.slug:
+                db.add(models.OldSlug(
+                    discussion=discussion,
+                    slug=discussion.slug,
+                    redirection_slug=slug))
+                # Redirecting all the old slugs to the newest slug
+                for old_slug in discussion.oldslug:
+                    old_slug.redirection_slug = slug
```

Comment:
@ayyazdaniaryan

Reply:
Yes, but definitely not here. You want to maintain data integrity. By doing this, you're overriding other redirections. You should do this logic in the view, when it is about to issue a path to render, instead of here.

Reply:
See my comment https://github.com/assembl/assembl/pull/642#discussion_r253912228